[convert-environment-variables-and-secrets-guide-stack-2] Convert env vars and secrets guide from repository to Definitions (#10793)
dagster-io/dagster: docs/content/guides/dagster/using-environment-variables-and-secrets.mdx

  - An environment variable, or
  - Provided directly in the configuration

As storing secrets in configuration is bad practice, we'll opt for using an environment variable. In this code, we're configuring the resource supplying it to our assets:

```python file=/guides/dagster/using_environment_variables_and_secrets/repository.py startafter=start endbefore=end
# __init__.py

from my_dagster_project import assets
from my_dagster_project.resources import github_api

from dagster import Definitions, load_assets_from_package_module

defs = Definitions(
    assets=load_assets_from_package_module(assets),
    resources={
        "github_api": github_api.configured(
            {"access_token": {"env": "GITHUB_ACCESS_TOKEN"}}
        )
    },
)
```

Let's review what's happening here:

- Using the <PyObject object="configured" /> method on the `github_api` resource, we can pass configuration info to the resource. In this example, we're telling Dagster to load the `access_token` from the `GITHUB_ACCESS_TOKEN` environment variable.

> - We're adding that configured resource to our `Definitions` object to that it is available for our assets.

Reviewer: ```suggestion
- We're adding that configured resource to our <PyObject object="Definitions" /> object so it's available for our assets.
```

Adding in the `PyObject` component here, so we can link to the API docs when they're ready

            }
        ),
    },
    "production": {
        "snowflake_io_manager": snowflake_pandas_io_manager.configured(
            {
                "account": "abc1234.us-east-1",
                "user": "system@company.com",
                "password": {"env": "SYSTEM_SNOWFLAKE_PASSWORD"},
                "database": "PRODUCTION",
                "schema": "HACKER_NEWS",
            }
        ),
    },
}

deployment_name = os.getenv("DAGSTER_DEPLOYMENT", "local")

defs = Definitions(
    assets=[items, comments, stories], resources=resources[deployment_name]
)
```

Let's review what's happening here:

- We've created a dictionary of resource definitions, `resources`, named after our `local` and `production` environments. In this example, we're using a [Pandas Snowflake I/O manager](/\_apidocs/libraries/dagster-snowflake-pandas).
- For both `local` and `production`, we used the <PyObject object="configured" /> method on `snowflake_pandas_io_manager` to provide environment-specific run configuration. Note the differences in configuration between `local` and `production`, specifically where environment variables were used.

> - Following the resources dictionary, we define the `deployment_name` variable, which determines the current executing environment. This variable defaults to `local`, ensuring that `DAGSTER_DEPLOYMENT=PRODUCTION` must be set to use the `production` configuration.

Reviewer: ```suggestion
- Following the `resources` dictionary, we define the `deployment_name` variable, which determines the current executing environment. This variable defaults to `local`, ensuring that `DAGSTER_DEPLOYMENT=PRODUCTION` must be set to use the `production` configuration.
```

Suggesting code formatting to explicitly refer to the `resources` key
